src/QueryRenderer.ts

```diff
@@ -273,25 +273,18 @@ class QueryRenderChild extends MarkdownRenderChild {
     }
 
     private async addGroupHeading(content: HTMLDivElement, group: GroupDisplayHeading) {
-        let header: any;
-        // Is it possible to remove the repetition here?
-        // Ideally, by creating a variable that contains h4, h5 or h6
-        // and then only having one call to content.createEl().
+        // Headings nested to 2 or more levels are all displayed with 'h6:
+        let header: keyof HTMLElementTagNameMap = 'h6';
         if (group.nestingLevel === 0) {
-            header = content.createEl('h4', {
-                cls: 'tasks-group-heading',
-            });
+            header = 'h4';
         } else if (group.nestingLevel === 1) {
-            header = content.createEl('h5', {
-                cls: 'tasks-group-heading',
-            });
-        } else {
-            // Headings nested to 2 or more levels are all displayed with 'h6:
-            header = content.createEl('h6', {
-                cls: 'tasks-group-heading',
-            });
+            header = 'h5';
```

Comment:
I've read the change several times, and each time I get to the h5 and wonder why there isn't an h6 there.

Text patterns in code are super important for readability.

In theory adding the 'h6' on the initialisation line is nice because it saves an else block...

In practice, being able to see the pattern of h4, h5 and h6 is more valuable, to make this PR even better still!

Reply:
Ah, looking at the history, I am effectively saying that the last-but-one commit was perfect!

Reply:
Not to worry, I'll go ahead and merge it.

Thanks for doing this!